src/bidiTab/bidiTab.ts

```diff
@@ -42,49 +47,33 @@ declare global {
 
     // Set from the server side if verbose logging is required.
     sendDebugMessage?: ((message: string) => void) | null;
-
-    // `window.setSelfTargetId` is called via `Runtime.evaluate` from the server side.
-    setSelfTargetId: (targetId: string) => void;
   }
 }
 
-// Initiate `setSelfTargetId` as soon as possible to prevent race condition.
-const waitSelfTargetIdPromise = waitSelfTargetId();
-
-void (async () => {
-  generatePage();
+generatePage();
+const mapperTabToServerTransport = new WindowBidiTransport();
+const cdpTransport = new WindowCdpTransport();
 
-  // Needed to filter out info related to BiDi target.
-  const selfTargetId = await waitSelfTargetIdPromise;
+/**
+ * Set `window.runMapper` to a function which launches the BiDi mapper instance.
+ * @param selfTargetId Needed to filter out info related to BiDi target.
+ */
+window.runMapper = async (selfTargetId) => {
+  console.log('launching with selfTargetId: ', selfTargetId);
 
-  const bidiServer = await BidiServer.createAndStart(
-    new WindowBidiTransport(),
+  await BidiServer.createAndStart(
+    mapperTabToServerTransport,
     /**
      * A CdpTransport implementation that uses the window.cdp bindings
      * injected by Target.exposeDevToolsProtocol.
      */
-    new CdpConnection(new WindowCdpTransport(), log),
+    new CdpConnection(cdpTransport, log),
     selfTargetId,
     new BidiParser(),
     log
   );
 
   log(LogType.debugInfo, 'Launched');
 
-  bidiServer.emitOutgoingMessage(
-    OutgoingMessage.createResolved({
-      launched: true,
-    }),
-    'launched'
-  );
-})();
-
-// Needed to filter out info related to BiDi target.
-async function waitSelfTargetId(): Promise<string> {
-  return new Promise((resolve) => {
-    window.setSelfTargetId = (targetId) => {
-      log(LogType.debugInfo, 'Current target ID:', targetId);
-      resolve(targetId);
-    };
-  });
-}
+  return 'launched';
```

Comment:
Ditto, please augment this log message; in case it's found somewhere out of context, it could be helpful.

Reply:
Removed not required return and extended the log message